subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/BuildInitPluginIntegrationTest.groovy

```diff
@@ -42,114 +39,184 @@ class BuildInitPluginIntegrationTest extends WellBehavedPluginTest {
+        ScriptDslFixture.assumeScriptDslSupportFor(targetScriptDsl, existingScriptDsl)
+
         given:
-        def customBuildScript = testDirectory.file("customBuild.gradle").createFile()
+        def targetDslFixture = ScriptDslFixture.of(targetScriptDsl, testDirectory)
+        def existingDslFixture = ScriptDslFixture.of(existingScriptDsl, testDirectory)
+
+        and:
+        def customBuildScript = existingDslFixture.scriptFile("customBuild").createFile()
 
         when:
         executer.usingBuildScript(customBuildScript)
-        run('init')
+        run('init', '--build-script-dsl', targetScriptDsl.id)
 
         then:
         result.assertTasksExecuted(":init")
-        result.output.contains("The build file 'customBuild.gradle' already exists. Skipping build initialization.")
+        result.output.contains("The build file '${customBuildScript.name}' already exists. Skipping build initialization.")
 
         and:
-        !buildFile.exists()
-        !settingsFile.exists()
-        wrapper.notGenerated()
+        !targetDslFixture.buildFile.exists()
+        !targetDslFixture.settingsFile.exists()
+        targetDslFixture.assertWrapperNotGenerated()
+
+        where:
+        existingScriptDsl | targetScriptDsl
+        GROOVY            | GROOVY
+        KOTLIN            | KOTLIN
+        GROOVY            | KOTLIN
+        KOTLIN            | GROOVY
     }
 
-    def "build file generation is skipped when part of a multi-project build with non-standard settings file location"() {
+    @Unroll
+    def "#targetScriptDsl build file generation is skipped when part of a multi-project build with non-standard #existingScriptDsl settings file location"() {
+        ScriptDslFixture.assumeScriptDslSupportFor(targetScriptDsl, existingScriptDsl)
+
         given:
-        def customSettings = testDirectory.file("customSettings.gradle")
+        def targetDslFixture = ScriptDslFixture.of(targetScriptDsl, testDirectory)
+        def existingDslFixture = ScriptDslFixture.of(existingScriptDsl, testDirectory)
+
+        and:
+        def customSettings = existingDslFixture.scriptFile("customSettings")
         customSettings << """
-include 'child'
+include("child")
 """
 
         when:
         executer.usingSettingsFile(customSettings)
-        run('init')
+        run('init', '--build-script-dsl', targetScriptDsl.id)
 
         then:
         result.assertTasksExecuted(":init")
         result.output.contains("This Gradle project appears to be part of an existing multi-project Gradle build. Skipping build initialization.")
 
         and:
-        !buildFile.exists()
-        !settingsFile.exists()
-        wrapper.notGenerated()
+        !targetDslFixture.buildFile.exists()
+        !targetDslFixture.settingsFile.exists()
+        targetDslFixture.assertWrapperNotGenerated()
+
+        where:
+        existingScriptDsl | targetScriptDsl
```

Comment:
⭕️ Could this duplication be removed with something like `[existingScriptDsl, targetScriptDsl] <<< ALL_DSL_COMBINATIONS`?

Reply:
Good call, this simplified things!